### PR TITLE
errno: move the error code from the parser/mysql to tidb/errno

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Some times we need to define/modify the error code.
The [error code](https://github.com/pingcap/parser/blob/master/mysql/errcode.go) is defined in the parser repo, so we need to update two repositories in a single PR, that's cumbersome.

I want to move the TiDB customed error back to the TiDB repo.

### What is changed and how it works?

The error code is now defined in github/pingcap/tidb/errno

I have a [PR](https://github.com/pingcap/parser/pull/764) to verify TiDB doesn't depend on parser/mysql errors anymore (it's not important to merge the PR or not).

From now on, don't add more error code to parser/mysql error definitions, use tidb/errno instead.
(We can choose to clean the parser repo or not)

### Check List 

Tests 

 - No code

Code changes

 - Has exported variable/fields change

Side effects

 ~~Breaking backward compatibility~~

Related changes

 - Need to cherry-pick to the release branch

It's better to cherry-pick it to reduce conflict in the future.

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @lysu @jackysp

**Comment by jackysp** [comment]
> Breaking backward compatibility
Why?

**Comment by tiancaiamao** [comment]
/run-all-tests

> > Breaking backward compatibility

Not really...

I mean the code compatibility, if we remove the parser exported function, the cherry-pick work would be trouble.

File: go.sum [hunk 1]
```diff
@@ -344,6 +344,12 @@ github.com/syndtr/goleveldb v0.0.0-20180815032940-ae2bd5eed72d h1:4J9HCZVpvDmj2t
 github.com/syndtr/goleveldb v0.0.0-20180815032940-ae2bd5eed72d/go.mod h1:Z4AUp2Km+PwemOoO/VB5AOx9XSsIItzFjoJlOSiYmn0=
 github.com/tiancaiamao/appdash v0.0.0-20181126055449-889f96f722a2 h1:mbAskLJ0oJfDRtkanvQPiooDH8HvJ2FBh+iKT/OmiQQ=
 github.com/tiancaiamao/appdash v0.0.0-20181126055449-889f96f722a2/go.mod h1:2PfKggNGDuadAa0LElHrByyrz4JPZ9fFx6Gs7nx7ZZU=
+github.com/tiancaiamao/parser v0.0.0-20200311024545-ff0568b33c35 h1:YXdyzP76zzpthmrHX+Ly8YB6pvn55leSvWEWYbimVik=
```
Comment (lysu) [inline go.sum]: why  introduce tiancaiamao/parser here?

Comment (tiancaiamao) [reply go.sum]: `go mod tidy` done

**Pull request merged**